src/terminal/scroll/stack.lua

```diff
@@ -0,0 +1,64 @@
+-- Terminal Scroll Stack Module.
+-- Manages a stack of scroll regions for terminal control.
+-- @module scroll.stack
+local M = {}
+M.output = require("terminal.output")
```

Comment:
this exports the `terminal.output` module again as `terminal.scroll.output`. That doesn't make sense.

```suggestion
local output = require("terminal.output")
```

and below change all `M.output...` to `output...`